packages/appservice-rest/src/Utilities/KuduServiceUtility.ts

```diff
@@ -37,15 +37,16 @@ export class KuduServiceUtility {
         return deploymentID;
     }
 
-    public async deployUsingZipDeploy(packagePath: string): Promise<string> {
+    public async deployUsingZipDeploy(packagePath: string, customMessage?: any): Promise<string> {
         try {
             console.log('Package deployment using ZIP Deploy initiated.');
 
             let queryParameters: Array<string> = [
                 'isAsync=true',
                 'deployer=' + GITHUB_ZIP_DEPLOY
             ];
-
+            var deploymentMessage = this._getUpdateHistoryRequest(null, null, customMessage).message;

```

Comment:
we should call this only if a customMessage is passed, not otherwise.

Reply:
this check is there inside this method
https://github.com/microsoft/pipelines-appservice-lib/blob/272b57f3993cba1edf87fff6bfcce730fdf167f9/packages/appservice-rest/src/Utilities/KuduServiceUtility.ts#L178

even if customMessage is not passed, this message will be added to logs
https://github.com/microsoft/pipelines-appservice-lib/blob/272b57f3993cba1edf87fff6bfcce730fdf167f9/packages/appservice-rest/src/Utilities/KuduServiceUtility.ts#L172